mautic/mautic
# [DPMMA-1660] Charts with email time stats in the contacts view

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [❌]
| New feature/enhancement? (use the a.x branch)      | [✅]
| Deprecations?                          | [❌]
| BC breaks? (use the c.x branch)        | [❌]
| Automated tests included?              | [✅] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

Charts with email activity in contact view (s/contacts/view/{id})

![contact_email_time](https://github.com/mautic/mautic/assets/102536220/93086f8b-6b67-49ca-906d-9a148f654970)
The charts show the contact's reading and clicking time preferences

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create one or more emails and send them to a contact.
3. Generate email opens and clicks from one contact, preferably at different times of the day and on different days.
4. Open contact view and check that the charts in "Best email time" tab show the correct data.
